saml2/saml2-service-provider/src/main/java/org/springframework/security/saml2/serviceprovider/authentication/Saml2Authentication.java

```diff
@@ -0,0 +1,57 @@
+/*
+ * Copyright 2002-2019 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.security.saml2.serviceprovider.authentication;
+
+import org.springframework.security.authentication.AbstractAuthenticationToken;
+import org.springframework.security.core.AuthenticatedPrincipal;
+import org.springframework.security.core.GrantedAuthority;
+import org.springframework.util.Assert;
+
+import java.util.Collection;
+
+/**
+ * @since 5.2
+ */
+public class Saml2Authentication extends AbstractAuthenticationToken {
+
+	private final String saml2Response;
+	private final AuthenticatedPrincipal principal;
+
+	public Saml2Authentication(String saml2Response, AuthenticatedPrincipal principal, Collection<? extends GrantedAuthority> authorities) {
```

Comment:
Change the order to be principal, saml2Response, authorities since principal, credentials is the order used in most of the other authentication objects

Reply:
hmm...it looks like samlResponse is still first as apposed to being second.